bounter/count_min_sketch.py

```diff
@@ -62,39 +52,42 @@ def __init__(self, size_mb=64, width=None, depth=None, algorithm='conservative')
 
         Args:
             size_mb (int): controls the maximum size of the Count-min Sketch table.
-                If width is provided, this parameter is ignored. Please note that the structure will use an
-                overhead of approximately 33 KB in addition to the table size.
+                If both width and depth is provided, this parameter is ignored.
+                Please note that the structure will use an overhead of approximately 65 KB in addition to the table size.
             depth (int): controls the number of rows of the table. Having more rows decreases probability of a big
-                overestimation but also linearly affects performance and table size. Choose a small number such as 4-8.
-                The algorithm will default depth 8 if width is provided. Otherwise, it will choose a depth in range 4-7
-                to best fill the maximum memory (for memory size which is a power of 2, depth of 4 is always used).
-            width (int): controls the number of hash buckets in one row of the table, overrides size_mb parameter.
+                overestimation but also linearly affects performance and table size. Choose a small number such as 6-10.
+                The algorithm will default depth 8 if width is provided. Otherwise, it will choose a depth in range 5-8
+                to best fill the maximum memory (for memory size which is a power of 2, depth of 8 is always used).
+            width (int): controls the number of hash buckets in one row of the table.
                 If width is not provided, the algorithm chooses the maximum width to fill the available size.
                 The more, the better, should be very large, preferably in the same order of magnitude as the cardinality
                 of the counted set.
             algorithm (str): controls the algorithm used for counting
-                - 'basic'
                 - 'conservative' (default)
-                - 'logcounter8'
-                - 'logcounter1024'
-                - 'logcons1024'
+                - 'log1024'
+                - 'log8'
         """
-        self.array_type = CountMinSketch.cell_type(algorithm)
-        cell_size = self.array_type().itemsize
+
+        cell_size = CountMinSketch.cell_size(algorithm);
+        self.cell_size_v = cell_size
+
+        if not isinstance(size_mb, int):
+            raise ValueError("size_mb must be an integer!")
+
         if width is None and depth is None:
-            self.width = 1 << (size_mb * 131072 // cell_size).bit_length()
-            self.depth = (size_mb * 1048576) // (self.width * cell_size)
+            self.width = 1 << (size_mb * (2 ** 16) // cell_size).bit_length()
+            self.depth = (size_mb * (2 ** 20)) // (self.width * cell_size)
```

Comment:
From the docstring, I would expect to see
```
self.depth = 8
```
here. No?

Reply:
I've clarified that a bit with:
```
self.width = 1 << (size_mb * (2 ** 20) // (cell_size * 8 * 2)).bit_length()
self.depth = (size_mb * (2 ** 20)) // (self.width * cell_size)
```
This will indeed result in 8 if size_mb is power of 2, but can produce depths 8-15 for other values (also fixed doc which incorrectly states 5-8, that was true for an older version)